core-bundle/contao/templates/twig/component/_list.html.twig

```diff
@@ -94,57 +95,59 @@
         {% block list_script %}
             {% add "list_script" to body %}
-                <script>
-                    {# Shuffle lists #}
-                    document.querySelectorAll('*[data-list-random]').forEach(list => {
-                        {# Shuffle order of children by appending them at a random position. #}
-                        for (let i = list.children.length; i >= 0; i--) {
-                            list.appendChild(list.children[Math.floor(Math.random() * i)]);
-                        }
-
-                        list.dataset.listRandom = 'ready';
-                    })
-
-                    {# Apply pagination #}
-                    document.querySelectorAll('*[data-list-paginate]').forEach(list => {
-                        const [items_per_page, max_items] = list.dataset.listPaginate.split(',').map(v => Number(v));
-                        const list_elements = list.querySelectorAll('li');
-                        const pagination = list.parentElement.querySelector('*[data-pagination]');
-                        const pagination_elements = pagination ? pagination.querySelectorAll('*[data-page]') : [];
-
-                        const show_list_element = (element, active) => {
-                            {% block pagination_script_show_element %}
-                                element.style.display = active ? 'revert' : 'none';
-                            {% endblock %}
-                        };
-
-                        const select_nav_element = (element, active) => {
-                            {% block pagination_script_activate_selector %}
-                                element.classList.toggle('active', active);
-                            {% endblock %}
-                        };
-
-                        const paginate = page => {
-                            list_elements.forEach((element, index) => {
-                                show_list_element(element, index >= ((page - 1) * items_per_page) && index < page * items_per_page && index < max_items);
-                            });
-                            pagination_elements.forEach(element => {
-                                select_nav_element(element, Number(element.dataset.page) === page)
-                            });
-                        };
-
-                        pagination_elements.forEach(pagination_element => {
-                            pagination_element.addEventListener('click', e => {
-                                e.preventDefault();
-                                paginate(Number(pagination_element.dataset.page));
-                            });
-                        });
-
-                        paginate(1);
-                        list.dataset.listPaginate = 'ready';
-                    });
-                </script>
+                {{ block('script_component') }}
             {% endadd %}
         {% endblock %}
     {% endif %}
 {% endblock %}
+
+{% block script_content %}
```

Comment:
This is unfortunate because now there is no more `<script>` tag and my IDE no longer recognizes the Javascript. 😢

Reply:
Yes, see also my description, I don't really like it either. If we agree that the new component does not really help all that much I can revert the changes and just use 

```twig
{% set script_attributes = attrs(script_attributes|default)
    .setIfExists('nonce', contao_csp_nonce('script-src'))
%}

<script{{ script_attributes }}>
```
everywhere directly.

Reply:
I would prefer this. 👍

Reply:
Me too.

Reply:
Done in 4fdd33f59107edbf4cc4a4ed74de93e365b97915